internal/bminventory/inventory.go

```diff
@@ -6977,7 +6977,17 @@ func (b *bareMetalInventory) refreshAfterUpdate(ctx context.Context, host *commo
 	err := b.hostApi.RefreshStatus(ctx, &host.Host, db)
 	if err != nil {
 		log.WithError(err).Errorf("Failed to refresh host %s, infra env %s during update", host.ID, host.InfraEnvID)
+		return err
+	}
+
+	if cluster != nil {
```

Comment:
i'm thinking about how to test it, does cluster refresh include the changes in the host? 
Maybe it need to run with `for update` option? 
@ori-amizur what do you think? 
A scenario here is a ready cluster with 3 masters, user change one of the hosts role into worker, cluster status will remain ready until the next iteration of the monitor. i think that this TX need a `for update` to include the changes in the host. 
Because in v2 host is separated from the cluster UI will need to get the cluster again after they apply a configuration on the host @jkilzi

Reply:
ack

Reply:
We need for update option on all entities that might be updated.  But in this case, the query must be inside the transaction